packages/dara-components/dara/components/smart/chat/endpoints.py

```diff
@@ -0,0 +1,16 @@
+"""
+Copyright (c) 2023 by Impulse Innovations Ltd. Private and confidential. Part of the causaLens product.
+"""
+
+import mimetypes
+
+from dara.core.http import post
+from dara.components.smart.chat.config import ChatConfig
+from dara.components.smart.chat.types import NewMessageBody
+
+mimetypes.init()
+
+
+@post(f'/chat/messages')
+def on_new_message(chat_config: ChatConfig, body: NewMessageBody):
+    return chat_config.on_new_message(body)
```

Comment:
Should probably wrap it in `run_user_handler` to make sure it can be async or sync.

Reply:
Good point, added